src/index/IndexMetaData.h

```diff
@@ -155,6 +155,8 @@ class IndexMetaData {
 
   size_t getVersion() const { return _version; }
 
+  MapType& data() { return _data; }
```

Comment:
In `std::vector` `data()` returns a pointer as it does in your `MmapVector` so why do you return a reference here?

Reply:
I think the probleme here is that `_data` is a complex type itsself. So I'd rather rename the `data()` function to `getDataRef` or something to not confuse it with the `data` of std::vector etc. which just have a raw pointer/ c-style-array as return value.